In [2]:
import numpy as np
import pandas as pd

# Tampilkan semua kolom
pd.set_option('display.max_columns', None)

In [3]:
base_dir = "C:\Purwadhika\Dataset\\"
path = base_dir + "Airbnb_Bangkok.csv"
df = pd.read_csv(path)
df.head()

<>:1: SyntaxWarning: invalid escape sequence '\P'
<>:1: SyntaxWarning: invalid escape sequence '\P'
C:\Users\Bintang\AppData\Local\Temp\ipykernel_37788\669406056.py:1: SyntaxWarning: invalid escape sequence '\P'
  base_dir = "C:\Purwadhika\Dataset\\"


Unnamed: 0      id                                            name  \
0           0   27934                 Nice room with superb city view   
1           1   27979                  Easy going landlord,easy place   
2           2   28745               modern-style apartment in Bangkok   
3           3   35780  Spacious one bedroom at The Kris Condo Bldg. 3   
4           4  941865                      Suite Room 3 at MetroPoint   

   host_id   host_name neighbourhood  latitude  longitude        room_type  \
0   120437      Nuttee   Ratchathewi  13.75983  100.54134  Entire home/apt   
1   120541         Emy       Bang Na  13.66818  100.61674     Private room   
2   123784  Familyroom     Bang Kapi  13.75232  100.62402     Private room   
3   153730     Sirilak     Din Daeng  13.78823  100.57256     Private room   
4   610315       Kasem     Bang Kapi  13.76872  100.63338     Private room   

   price  minimum_nights  number_of_reviews last_review  reviews_per_month  \
0   1905               3                 65  2020-01-06               0.50   
1   1316               1                  0         NaN                NaN   
2    800              60                  0         NaN                NaN   
3   1286               7                  2  2022-04-01               0.03   
4   1905               1                  0         NaN                NaN   

   calculated_host_listings_count  availability_365  number_of_reviews_ltm  
0                               2               353                      0  
1                               2               358                      0  
2                               1               365                      0  
3                               1               323                      1  
4                               3               365                      0

In [4]:
df.head(10)

Unnamed: 0       id                                            name  \
0           0    27934                 Nice room with superb city view   
1           1    27979                  Easy going landlord,easy place   
2           2    28745               modern-style apartment in Bangkok   
3           3    35780  Spacious one bedroom at The Kris Condo Bldg. 3   
4           4   941865                      Suite Room 3 at MetroPoint   
5           5  1704776                  NEw Pro!!  Bungalow Bkk Centre   
6           6    48736                 Condo with Chaopraya River View   
7           7  1738669                 1 chic bedroom apartment in BKK   
8           8  1744248              Batcave, Pool view, near Chatuchak   
9           9   952677                   Standard Room Decor do Hostel   

   host_id   host_name  neighbourhood  latitude  longitude        room_type  \
0   120437      Nuttee    Ratchathewi  13.75983  100.54134  Entire home/apt   
1   120541         Emy        Bang Na  13.66818  100.61674     Private room   
2   123784  Familyroom      Bang Kapi  13.75232  100.62402     Private room   
3   153730     Sirilak      Din Daeng  13.78823  100.57256     Private room   
4   610315       Kasem      Bang Kapi  13.76872  100.63338     Private room   
5  2129668    Wimonpak  Bang Kho laen  13.69757  100.52880  Entire home/apt   
6   222005    Athitaya     Rat Burana  13.68556  100.49535     Private room   
7  7045870    Jiraporn     Chatu Chak  13.82925  100.56737  Entire home/apt   
8  9181769         Nol     Chatu Chak  13.81693  100.56433  Entire home/apt   
9  5171292      Somsak     Khlong San  13.72040  100.50757     Private room   

   price  minimum_nights  number_of_reviews last_review  reviews_per_month  \
0   1905               3                 65  2020-01-06               0.50   
1   1316               1                  0         NaN                NaN   
2    800              60                  0         NaN                NaN   
3   1286               7                  2  2022-04-01               0.03   
4   1905               1                  0         NaN                NaN   
5   1000             250                 19  2017-08-03               0.17   
6   1558               3                  1  2014-02-03               0.01   
7   1461               1                  0         NaN                NaN   
8   1558               3                 10  2016-03-29               0.09   
9    700               2                  4  2019-12-27               0.03   

   calculated_host_listings_count  availability_365  number_of_reviews_ltm  
0                               2               353                      0  
1                               2               358                      0  
2                               1               365                      0  
3                               1               323                      1  
4                               3               365                      0  
5                              41               365                      0  
6                               1               365                      0  
7                               1               365                      0  
8                               1               365                      0  
9                              10                87                      0

### 📘 Data Dictionary
Penjelasan setiap kolom dapat dibaca pada dictionary ini:

| Feature                         | Description                                                                                                                                     |
|---------------------------------|-------------------------------------------------------------------------------------------------------------------------------------------------|
| id                              | Airbnb's unique identifier for the listing.                                                                                                    |
| name                            | Name of the listing.                                                                                                                           |
| host_id                         | Airbnb's unique identifier for the host/user.                                                                                                  |
| host_name                       | Name of the host. Usually, just the first name(s).                                                                                             |
| neighborhood                    | The neighborhood is geocoded using the latitude and longitude against neighborhoods as defined by open or public digital shapefiles.          |
| latitude                        | Uses the World Geodetic System (WGS84) projection for latitude and longitude.                                                                  |
| longitude                       | Uses the World Geodetic System (WGS84) projection for latitude and longitude.                                                                  |
| room_type                       | [Entire home/apt | Private room | Shared room | Hotel] - Classification of the type of space offered.                                          |
| price                           | Daily price in local currency. Note: The `$` sign may be used despite the locale.                                                              |
| minimum_nights                  | The minimum number of night stays for the listing (calendar rules may differ).                                                                |
| number_of_reviews               | The number of reviews the listing has.                                                                                                         |
| last_review                     | The date of the last/newest review.                                                                                                            |
| calculated_host_listings_count | The number of listings the host has in the current scrape in the city/region geography.                                                        |
| availability_365                | The calendar determines the availability of the listing x days in the future. A listing may be unavailable due to bookings or host blocks.   |
| number_of_reviews_ltm          | The number of reviews the listing has received in the last 12 months.                                                                          |

## Data Explroation
Pada tahap ini kita akan mencari tau karakteristik dari dataset yang digunakan untuk analisis Data Airbnb Bangkok

In [27]:
df.query("last_review.isna()")["number_of_reviews"].unique()

array([0], dtype=int64)

In [5]:
# Menampilkan karakteristik tiap kolom dalam dataframe
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 15854 entries, 0 to 15853
Data columns (total 17 columns):
 #   Column                          Non-Null Count  Dtype  
---  ------                          --------------  -----  
 0   Unnamed: 0                      15854 non-null  int64  
 1   id                              15854 non-null  int64  
 2   name                            15846 non-null  object 
 3   host_id                         15854 non-null  int64  
 4   host_name                       15853 non-null  object 
 5   neighbourhood                   15854 non-null  object 
 6   latitude                        15854 non-null  float64
 7   longitude                       15854 non-null  float64
 8   room_type                       15854 non-null  object 
 9   price                           15854 non-null  int64  
 10  minimum_nights                  15854 non-null  int64  
 11  number_of_reviews               15854 non-null  int64  
 12  last_review                     

Berdasarkan informasi diatas, terdapat **16 kolom** dengan **jumlah baris 15.853** secara keseluruhan. Kita asumsikan bahwa total baris adalah jumlah penginapan untuk airbnb di thailand. terdapat kolom `Unnamed: 0` yang tidak diperlukan

Pada beberapa kolom terdapat missing value seperti `name` dengan **8 missing values**, `last_review` dengan **5790 missing values** dan `reviews_per_month` dengan **5790 missing values**.
sehingga kita perlu melakukan cleaning data tersebut untuk menghindari bias pada saat analisis

In [6]:
# Deskriptif analytics pada kolom number
df.describe()

Unnamed: 0            id       host_id      latitude     longitude  \
count  15854.000000  1.585400e+04  1.585400e+04  15854.000000  15854.000000   
mean    7926.500000  1.579397e+17  1.541058e+08     13.745144    100.559903   
std     4576.799919  2.946015e+17  1.318726e+08      0.043040      0.050911   
min        0.000000  2.793400e+04  5.892000e+04     13.527300    100.329550   
25%     3963.250000  2.104509e+07  3.974431e+07     13.720090    100.529690   
50%     7926.500000  3.503734e+07  1.224556e+08     13.738490    100.561415   
75%    11889.750000  5.256154e+07  2.390547e+08     13.759497    100.585150   
max    15853.000000  7.908162e+17  4.926659e+08     13.953540    100.923440   

              price  minimum_nights  number_of_reviews  reviews_per_month  \
count  1.585400e+04    15854.000000       15854.000000       10064.000000   
mean   3.217704e+03       15.292355          16.654157           0.813145   
std    2.497212e+04       50.815020          40.613331           1.090196   
min    0.000000e+00        1.000000           0.000000           0.010000   
25%    9.000000e+02        1.000000           0.000000           0.120000   
50%    1.429000e+03        1.000000           2.000000           0.435000   
75%    2.429000e+03        7.000000          13.000000           1.060000   
max    1.100000e+06     1125.000000        1224.000000          19.130000   

       calculated_host_listings_count  availability_365  number_of_reviews_ltm  
count                    15854.000000      15854.000000           15854.000000  
mean                        13.889618        244.378643               3.481519  
std                         30.269848        125.843224               8.916937  
min                          1.000000          0.000000               0.000000  
25%                          1.000000        138.000000               0.000000  
50%                          4.000000        309.000000               0.000000  
75%                         13.000000        360.000000               3.000000  
max                        228.000000        365.000000             325.000000

Data deksriptif diatas menunjukan secara singkat karakteristik dari kolom number. Berdasarkan data diatas kita bisa mengetahui sebaran harga pada layanan Airbnb bangkok pada kolom `Price`

#### 1. Analisis data User
- Berapa jumlah masing-masing kamar pada tiap room_type?
- type_room apa yang paling banyak berdasarkan persentase?

In [7]:
df['room_type'].nunique()

4

In [8]:
df['room_type'].unique()

array(['Entire home/apt', 'Private room', 'Hotel room', 'Shared room'],
      dtype=object)

In [9]:
df.loc[df['room_type'] == 'Entire home/apt', 'name'].nunique()

8239

In [10]:
df.loc[df['room_type'] == 'Private room', 'name'].nunique()

5524

In [11]:
df.loc[df['room_type'] == 'Hotel room', 'name'].nunique()

616

In [12]:
df.loc[df['room_type'] == 'Shared room', 'name'].nunique()

453

In [13]:
room_type_counts = df['room_type'].value_counts()
room_type_nunique = df['room_type'].nunique()
room_type_percentage = (room_type_counts / len(df)) * 100
print(room_type_percentage)
print(f"Jumlah unique room_type: {room_type_nunique}")

room_type
Entire home/apt    56.212943
Private room       36.394601
Hotel room          4.093604
Shared room         3.298852
Name: count, dtype: float64
Jumlah unique room_type: 4


Berdasarkan data diatas secara persentase adalah tipe ruangan yang berupa rumah atau apartemen dengan persentase 56% atau dengan jumlah 8239

In [14]:
# Cari host dengan jumlah listing terbanyak berdasarkan kolom 'calculated_host_listings_count'
max_listing_count = df['calculated_host_listings_count'].max()
top_hosts = df[df['calculated_host_listings_count'] == max_listing_count][['host_name', 'name', 'calculated_host_listings_count']]

print(top_hosts)

      host_name                                               name  \
6473      Curry    Near The Grand Palace/ASOK Station/Sukhumvit#05   
6479      Curry  Near The Grand Palace/ASOK Station/Sukhumvit 5...   
6672      Curry    Near The Grand Palace/ASOK Station/Sukhumvit#06   
7009      Curry         New! Sukhumvit NANA 1BR Suit near BTS 2PPL   
7244      Curry  30days!  Sukhumvit NANA spacious 1BR 2PPL near...   
...         ...                                                ...   
14906     Curry        New!Near The Grand Palace/ASOK Station/4PPL   
15336     Curry  30days! AirportLink Sukhumvit NANA MaxValu 2BR...   
15474     Curry  月租特惠！豪华两卧两卫！Ekkamai BTS/The Emporium/曼谷大学/无边泳池...   
15515     Curry  One Siam/Platinum Fashion Mall/Mrt Arl Bts/MBK...   
15783     Curry     5days!New!Bang Phlat MRT 600m/swiming pool/gym   

       calculated_host_listings_count  
6473                              228  
6479                              228  
6672                              228  

#### 2. Anlisis lokasi properti yang populer

In [15]:
popular_neighbourhoods = df['neighbourhood'].value_counts().reset_index()
popular_neighbourhoods.columns = ['neighbourhood', 'number_of_listings']
popular_neighbourhoods.head(10)  # 10 lokasi terpopuler

neighbourhood  number_of_listings
0       Vadhana                2153
1   Khlong Toei                2097
2   Huai Khwang                1125
3   Ratchathewi                1114
4      Bang Rak                 827
5        Sathon                 809
6   Phra Nakhon                 619
7  Phra Khanong                 558
8    Chatu Chak                 548
9       Bang Na                 458

Berdasarkan data analisis properti yang populer didapatkanlah rata-rata harga berdasarkan daerah dengan banyaknya jumlah properti seperti dibawah ini


In [16]:
# Membuat kolom baru 'avg_price' pada popular_neighbourhoods berdasarkan rata-rata price di df
avg_price_per_neighbourhood = df.groupby('neighbourhood')['price'].mean().reset_index()
avg_price_per_neighbourhood.columns = ['neighbourhood', 'avg_price']

# Merge dengan popular_neighbourhoods
popular_neighbourhoods = popular_neighbourhoods.merge(avg_price_per_neighbourhood, on='neighbourhood')
popular_neighbourhoods.head(10)

neighbourhood  number_of_listings    avg_price
0       Vadhana                2153  4452.131444
1   Khlong Toei                2097  4249.329995
2   Huai Khwang                1125  5036.084444
3   Ratchathewi                1114  3199.271993
4      Bang Rak                 827  3056.336155
5        Sathon                 809  2338.238566
6   Phra Nakhon                 619  1731.361874
7  Phra Khanong                 558  1483.254480
8    Chatu Chak                 548  2294.386861
9       Bang Na                 458  2060.000000

In [17]:
# Cek missing Value untuk semua kolom
missing_value_percentage = df.isna().sum() / len(df) * 100

missing_summary = pd.DataFrame({
                                'Missing_count': df.isna().sum(),
                                'Missing_Percentage': missing_value_percentage.round(2)
})

In [18]:
missing_summary.sort_values('Missing_Percentage', ascending=False)

Missing_count  Missing_Percentage
reviews_per_month                        5790               36.52
last_review                              5790               36.52
name                                        8                0.05
host_name                                   1                0.01
Unnamed: 0                                  0                0.00
minimum_nights                              0                0.00
availability_365                            0                0.00
calculated_host_listings_count              0                0.00
number_of_reviews                           0                0.00
room_type                                   0                0.00
price                                       0                0.00
id                                          0                0.00
longitude                                   0                0.00
latitude                                    0                0.00
neighbourhood                               0                0.00
host_id                                     0                0.00
number_of_reviews_ltm                       0                0.00

In [25]:
df['reviews_per_month'].describe()

count    10064.000000
mean         0.813145
std          1.090196
min          0.010000
25%          0.120000
50%          0.435000
75%          1.060000
max         19.130000
Name: reviews_per_month, dtype: float64